City-of-Helsinki/drupal-helfi-rekry
# UHF-5307: Job listings permission

# [UHF-5307](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5307)
Makes it possible for HR to unpublish job listings.

## What was done
* Added HR role which has permission to change job listings publish status and prevent publish field.
  * The other job listings content fields are disabled.
* Small changes for other roles regarding job listing permissions.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git pull origin UHF-5307-job-listings-permission`
* Run `make drush-cim`
* Run `make drush-cr`
* Run `make shell` and `drush migrate:import helfi_rekry_jobs:all` to import the job listings if not previously imported.

## How to test
* Create a test user with HR role.
* Log in using the new user account.
* After importing job listings, try to edit one.
  * The user should be able to edit the "Prevent publishing" field and also changing the publish/un-publish status. (Also, the schedule options are visible so that the user can see possible error messages related to those dates.)
  * It should not be possible to edit the other content fields (including title).
  * The user can't edit other content types.

## Designers review
